plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -410,6 +411,35 @@ private static <K, V> V getWithValueHolder(Cache<K, AtomicReference<V>> cache, K
         return value;
     }
 
+    private static <K, V> V getWithValueHolder(
+            Cache<K, AtomicReference<V>> cache, K key,
+            Predicate<V> shouldReload,
+            Supplier<V> loader,
+            Function<V, V> reloader,
```

Comment:
- you do not need shouldReload & reloader as separate things.
the reloader can be no-op when  nothing to reload

- "reloader" sounds like "loading what we already know, but afresh", while what you mean is "load additional information  on top of what we already have". maybe `incrementalLoader`?

Reply:
> you do not need shouldReload & reloader as separate things.

i see it makes sense to keep them separate because this aligns better with the bulk version used for part stats

perhaps rename "shouldReload" to "is sufficient"